src/api_keys.rs

```diff
@@ -139,21 +166,27 @@ pub mod types {
         pub data: Vec<ApiKey>,
     }
 
+    /// Name and ID of an existing API key.
     #[must_use]
     #[derive(Debug, Clone, Deserialize)]
     pub struct ApiKey {
         /// The ID of the API key.
-        pub id: String,
+        pub id: ApiKeyId,
         /// The name of the API key.
         pub name: String,
+
         /// The date and time the API key was created.
+        #[cfg(not(feature = "time"))]
         pub created_at: String,
+        #[cfg(feature = "time")]
+        #[serde(with = "time::serde::iso8601")]
+        pub created_at: time::OffsetDateTime,
```

Comment:
This is up for discussion but I feel like having these be *not* mutually exclusive would be better here. [The official docs don't really like mutually exclusive features](https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features) and I think for most stuff other than sync/async splits they are not worth it. The crate's code becomes a bit messier but more importantly, if someone is using this crate without the `time` feature initially and they later on realize that it would be neat to have the relevant types, it is possible that some of their code now fails to compile if they used the fact that the field was a string before.

Reply:
what name should it use? `created_at: String` and `created_at_time: time::OffsetDateTime`?

Reply:
Yeah I also tried thinking of what to name it but didn't come up with something non-ugly. Maybe we could do `created_at_datetime`? That at least feels like a type. Not sure if this needs custom (de)serialize traits, I've always used chrono instead of time but I would assume time has methods for parsing dates (?)

Reply:
Yeah, time has these methods, but then we'd need to `#[serde(skip)] created_at` and manually parse the `created_at_time` after calling `response.json()`.

Reply:
Ah I see what you mean. [Doesn't the `serde` feature make this work though](https://time-rs.github.io/book/how-to/parse-dates.html#parsing-into-structs-with-serde)?

Reply:
it does work, but the "created_at" field consumes that part of the response json, and afaik it's not possible to parse the same value twice without impl custom serialize/deserialize traits. at least it used to be this way.

Reply:
Oh dang I see., that's annoying. I think implementing deserialize manually would be the most convenient for anyone using this crate. We could also put the parsed time behind a function instead of a field but that would be weird since everything else is a field (+ that would need to be parsed on each function call, unless we used `OnceLock` or something similar).

Reply:
Actually it might be possible to use [this](https://serde.rs/container-attrs.html#from)

Reply:
or we could just get rid of a feature and make it `OffsetDateTime`  by default

Reply:
Would something like this work?

```rs
    #[must_use]
    #[derive(Debug, Clone, Deserialize)]
    #[serde(from = "ApiKeyRaw")]
    pub struct ApiKey {
        /// The ID of the API key.
        pub id: ApiKeyId,
        /// The name of the API key.
        pub name: String,

        /// The date and time the API key was created.
        pub created_at: String,
        #[cfg(feature = "time")]
        pub created_at_time: time::OffsetDateTime,
    }

    impl From<ApiKeyRaw> for ApiKey {
        fn from(value: ApiKeyRaw) -> Self {
            todo!()
        }
    }

    #[derive(Debug, Clone, Deserialize)]
    struct ApiKeyRaw {
        pub id: ApiKeyId,
        pub name: String,
        pub created_at: String,
    }
```

Reply:
Quick and dirty `From` implementation

```rs
    impl From<ApiKeyRaw> for ApiKey {
        fn from(value: ApiKeyRaw) -> Self {
            ApiKey { 
                id: value.id, 
                name: value.name, 
                created_at: value.created_at.clone(), 
                created_at_time: time::OffsetDateTime::parse(&value.created_at, &Iso8601::DEFAULT).unwrap()
            }
        }
    }
```

Reply:
The test in that file passes for me so I'm guessing it works 😅 

Let me know what you think

Reply:
I think so, but that would be Email, Audience, Contact, Domain, ApiKey and DomainReply (idk about this one) duplicates.
The only reason I'd like to have a parsed datetime is to make it easier to navigate logs. Having the date repeated doesn't really add much value in this context.

Reply:
Btw, should we remove DomainReply? The only difference seems to be the 
```rust
#[serde(rename = "dnsProvider")]
pub dns_provider: String,
```
field, it could be replaced it with `Domain` with additional `pub dns_provider: String,` that would be present only as a response of a `Domain::create`.

Reply:
> I think so, but that would be Email, Audience, Contact, Domain, ApiKey and DomainReply (idk about this one) duplicates.

Well the `ApiKeyRaw` would be just an internal type not exposed anywhere so for the users this wouldn't be tedious. For this crate I mean it's a bit annoying but it is not that bad.

Reply:
For the domain reply, I think it is fine now, especially since all our reply types have `Reply` in the type name so it would be confusing for this to be an exception

Reply:
I think it's the only `*Reply` left 😄

Reply:
I've lost track of some of the changes at this point 🤣 makes sense then haha

Reply:
Yeah, I've renamed most of the public `Method*Request` and `Method*Response` classes. Some were reduced to a single argument (e.g. `CreateAudienceRequest`), while others were incorrectly generated by an open-api generator (e.g. `ListAudiencesItemResponse`, which was supposed to be `GetAudienceResponse`).

Reply:
Any thoughts on the time parsing?

Reply:
Not a big fan of seeing the same thing repeated twice (both `created_at*` and `*Raw`), but I'm fine with that.

Reply:
Do you think it would be better to manually implement deserialize instead?

Reply:
We'd still have `created_at`/`created_at_time`. Pretty much the same thing (if you print it). 
And it's the same ~200 loc, just more complicated.

Reply:
Ah I see what you mean. I still feel like having that duplication is better than mutually excluding them tho

Reply:
Agree, that's why I'd rather avoid both issues by either removing time support or by making it the default.

Reply:
I think removing it all together would be reasonable as well, considering we are providing the string date anyway, anyone that wants to parse it can use whichever crate they want to do so. Maybe that's the best solution to be honest

Reply:
Done